python/mcap/mcap0/records.py

```diff
@@ -0,0 +1,383 @@
+from dataclasses import dataclass, field
+from typing import Dict, List, Tuple
+
+from .data_stream import ReadDataStream, WriteDataStream
+from .opcode import Opcode
+
+
+@dataclass
+class McapRecord:
+    def write(self, stream: WriteDataStream):
+        pass
+
+
+@dataclass
+class Attachment(McapRecord):
+    name: str
+    created_at: int
+    log_time: int
+    content_type: str
+    data: bytes
+
+    def write(self, stream: WriteDataStream):
+        stream.start_record(Opcode.ATTACHMENT)
+        stream.write_prefixed_string(self.name)
+        stream.write8(self.created_at)
+        stream.write8(self.log_time)
+        stream.write_prefixed_string(self.content_type)
+        stream.write8(len(self.data))
+        stream.write(self.data)
+        stream.write4(0)  # TODO: crc
+        stream.finish_record()
+
+    @staticmethod
+    def read(stream: ReadDataStream):
+        name = stream.read_prefixed_string()
+        created_at = stream.read8()
+        log_time = stream.read8()
+        content_type = stream.read_prefixed_string()
+        data_length = stream.read8()
+        data = stream.read(data_length)
+        _crc = stream.read4()
+        return Attachment(
+            name=name,
+            created_at=created_at,
+            log_time=log_time,
+            content_type=content_type,
+            data=data,
+        )
+
+
+@dataclass
+class AttachmentIndex(McapRecord):
+    offset: int
+    length: int
+    log_time: int
+    data_size: int
+    name: str
+    content_type: str
+
+    @staticmethod
+    def read(stream: ReadDataStream):
+        offset = stream.read8()
+        length = stream.read8()
+        log_time = stream.read8()
+        data_size = stream.read8()
+        name = stream.read_prefixed_string()
+        content_type = stream.read_prefixed_string()
+        return AttachmentIndex(
+            offset=offset,
+            length=length,
+            log_time=log_time,
+            data_size=data_size,
+            name=name,
+            content_type=content_type,
+        )
+
+
+@dataclass
+class ChannelInfo(McapRecord):
+    id: int = field(metadata={"value_type": ["int"]})
```

Comment:
why do we use this `metadata` thing for ints but not strings?

Reply:
We only have one type of string to handle but need to handle different ints differently. This will go away in the json version.